src/main/java/core/basesyntax/service/impl/DataValidatorImpl.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dto.FruitTransactionDto;
+import core.basesyntax.exception.InvalidFruitTypeException;
+import core.basesyntax.exception.InvalidOperationException;
+import core.basesyntax.exception.InvalidQuantityException;
+import core.basesyntax.service.DataValidator;
+
+public class DataValidatorImpl implements DataValidator {
+    private static final String BALANCE = FruitDataParser.Operation.BALANCE.getCode();
+    private static final String SUPPLY = FruitDataParser.Operation.SUPPLY.getCode();
+    private static final String RETURN = FruitDataParser.Operation.RETURN.getCode();
+    private static final String PURCHASE = FruitDataParser.Operation.PURCHASE.getCode();
+    private static final String SPECIAL_CHARACTERS = ".*[$&+,/:;=?@#|'<>.-^*()%!].*";

```

Comment:
what will be if your String will contain `_`? let's better check that string contains only numbers